internal/services/bot/bot_channel_directline_resource.go

```diff
@@ -236,6 +264,11 @@ func resourceBotChannelDirectlineUpdate(d *pluginsdk.ResourceData, meta interfac
 		return fmt.Errorf("updating Directline Channel for Bot %q (Resource Group %q): %+v", id.BotServiceName, id.ResourceGroup, err)
 	}
 
+	// Unable to add a new site with enhanced_authentication_enabled, block_user_upload_enabled, endpoint_parameters_enabled, no_storage_enabled in the same operation, so we need to make two calls
```

Comment:
There is no difference between this Update and the one above, do you have a specific reason to do a second update here?

Meanwhile, the whole `resourceBotChannelDirectlineUpdate` seems not checking whether a specific property is changed or not, during its construction of the request body. Shall we follow the get-then-put fasion, to only update the properties that are changed, so that we can support the `ignore_changes` better? We can do this either in this PR or in another PR.

Reply:
The first update only can add a new site but it cannot set the value for it's sub properties. The sub properties need to be set in the second update.

Reply:
This is really weired... Can we create a ticket to the service side to ask them improve this (given this shouldn't be a breaking change)?

Reply:
Sure. I filed an issue on https://github.com/Azure/azure-rest-api-specs/issues/25758